### ISSUE TITLE
Task name normalization for log folders: special characters in task names (e.g. ()) lead to Job Execution error

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-server#1643]
@adrtsc discovered an important issue in Fractal server 2.3.0 that we did not test for thoroughly enough:

As of https://github.com/fractal-analytics-platform/fractal-server/pull/1523 (part of the 2.2.0 release we hadn't broadly deployed until the 2.3.0 update), we group all the log files, sbatch files etc. in folders that contain the task name. But we did not take care of handling special characters in task names reliably. And on a bash console, the () would need an escape character (\) to be interpreted correctly.

This bug is breaking existing workflows and we should deploy a bugfix for it as soon as possible.

Special characters in task names appear to be handles robustly across very many parts of Fractal server, just having them show up as folder names leads to issues with the slurm srun commands. I think we can come up with a taskname normalization strategy for those log/sbatch folder names.

The only special character that shows up in task names so far and appears to make an issue are (). There are also some `-` (in the same task), but I assume they won't make an issue. But it would be good to protect against other special characters future tasks developers may introduce.

-------
### Known Tasks affected
In Fractal V2:
- Calculate Registration (image-based)

In Fractal V1:
- Create OME-ZARR structure (multiplexing)

-------

### Details

What then happens is that any task with special characters always triggers a Job Execution Error:
Example task:
Calculate Registration (image-based)

It's put into the folder: `/path/to/artifacts/proj_v2_0000036_wf_0000041_job_0000131_20240713_195124/1_calculate_registration_(image-based)`

And the srun command is something like:
```
srun --ntasks=1 --cpus-per-task=$SLURM_CPUS_PER_TASK --mem=1000MB /path/to/fractal-server-env/bin/python -m fractal_server.app.runner.executors.slurm.remote --input-file /path/to/artifacts/proj_v2_0000036_wf_0000041_job_0000131_20240713_195124/1_calculate_registration_(image-based)/1_in_3a4PuCvItfFuHKVreAw2JpJbhFO7vh9s.pickle --output-file /path/to/.fractal-cache/proj_v2_0000036_wf_0000041_job_0000131_20240713_195124/1_calculate_registration_(image-based)/1_out_3a4PuCvItfFuHKVreAw2JpJbhFO7vh9s.pickle &
```

The problem here being paths like: `/path/to/artifacts/proj_v2_0000036_wf_0000041_job_0000131_20240713_195124/1_calculate_registration_(image-based)/1_in_3a4PuCvItfFuHKVreAw2JpJbhFO7vh9s.pickle`

For such a path to work on bash, we'd need to use an escape character (\) before the special character (.

This then leads to the following job execution error:
```
Content of /path/to/artifacts/proj_v2_0000036_wf_0000041_job_0000131_20240713_195124/1_calculate_registration_(image-based)/1_slurm_10446035.err:
/var/lib/slurm/slurmd/job10446035/slurm_script: 19: /var/lib/slurm/slurmd/job10446035/slurm_script: Syntax error: "(" unexpected
```

**Comment by tcompa** [comment]
Now fixed in version 2.3.1 (as part of work on #1645).

**Comment by tcompa** [comment]
@adrtsc confirmed that he could run one of these workflows again, on an instance with v2.3.2
